editor/src/clj/editor/debug_view.clj

```diff
@@ -473,38 +474,39 @@
 
 (def ^:private mobdebug-port 8172)
 
-(defn- show-connect-failed-dialog! [target-address port ^Exception exception]
-  (let [msg (str "Failed to attach debugger to " target-address ":" port ".\n"
-                 "Check that the game is running and is reachable over the network.\n")]
-    (log/error :msg msg :exception exception)
-    (dialogs/make-info-dialog
-      {:title "Attach Debugger Failed"
-       :icon :icon/triangle-error
-       :header msg
-       :content (.getMessage exception)})))
+(defn- show-connect-failed-info! [target-address port ^Exception exception workspace]
+  (ui/run-later
+    (let [msg (str "Failed to attach debugger to " target-address ":" port ".\n"
+                   "Check that the game is running and is reachable over the network.\n"
+                   "Error message:\n" (.getMessage exception))]
+      (log/error :msg msg :exception exception)
+      (notifications/show!
+        (workspace/notifications workspace)
+        {:type :error
+         :id ::debugger-connection-error
+         :text msg}))))
 
 (defn start-debugger!
-  [debug-view project target-address instance-index]
-  (try
-    (mobdebug/connect! target-address (+ mobdebug-port instance-index)
-                       (fn [debug-session]
-                         (ui/run-now
-                           (g/update-property! debug-view :debug-session
-                                               (fn [old new]
-                                                 (when old (mobdebug/close! old))
-                                                 new)
-                                               debug-session)
-                           (update-breakpoints! debug-session (g/node-value project :breakpoints))
-                           (mobdebug/run! debug-session (make-debugger-callbacks debug-view))
-                           (state-changed! debug-view true)))
-                       (fn [_debug-session]
-                         (ui/run-now
-                           (g/set-property! debug-view
-                                            :debug-session nil
-                                            :suspension-state nil)
-                           (state-changed! debug-view false))))
-    (catch Exception exception
-      (show-connect-failed-dialog! target-address (+ mobdebug-port instance-index) exception))))
+  [debug-view project target-address instance-index workspace]
```

Comment:
We could get the `workspace` using `(project/workspace project)`, so we don't need to add it as an argument here.

Reply:
fixed